src/poliastro/twobody/orbit.py

```diff
@@ -681,6 +681,99 @@ def geostationary(
         altitude = geo_radius - attractor.R
         return cls.circular(attractor, altitude)
 
+    @classmethod
+    def heliosynchronous(
+        cls,
+        a=None,
+        ecc=None,
+        inc=None,
+        raan=0 * u.deg,
+        argp=0 * u.deg,
+        nu=0 * u.deg,
+        epoch=J2000,
+        plane=Planes.EARTH_EQUATOR,
+    ):
+        r""" Solves for a Sun-Synchronous orbit. These orbits make use of the J2
+        perturbation to precess in order to be always towards Sun. At least
+        two parameters of the set {a, ecc, inc} are needed in order to solve
+        for these kind of orbits. Relationships among them are given by:
+
+        .. math::
+            \begin{align}
+                a &= \left (\frac{-3R_{\bigoplus}J_{2}\sqrt{\mu}\cos(i)}{2\dot{\Omega}(1-e^2)^2}  \right ) ^ {\frac{2}{7}}\\
+                e &= \sqrt{1 - \sqrt{\frac{-3R_{\bigoplus}J_{2}\sqrt{\mu}cos(i)}{2a^{\frac{7}{2}}\dot{\Omega}}}}\\
+                i &= \arccos{\left ( \frac{-2a^{\frac{7}{2}}\dot{\Omega}(1-e^2)^2}{3R_{\bigoplus}J_{2}\sqrt{\mu}} \right )}\\
+            \end{align}
+
+        Parameters
+        ----------
+        a: ~astropy.units.Quantity
+            Semi-major axis.
+        ecc: ~astropy.units.Quantity
+            Eccentricity.
+        inc: ~astropy.units.Quantity
+            Inclination.
+        raan : ~astropy.units.Quantity
+            Right ascension of the ascending node.
+        argp : ~astropy.units.Quantity
+            Argument of the pericenter.
+        nu : ~astropy.units.Quantity
+            True anomaly.
+        epoch : ~astropy.time.Time, optional
+            Epoch, default to J2000.
+        plane : ~poliastro.frames.Planes
+            Fundamental plane of the frame.
+        """
+
+        # Constants for Sun-Synchronours Orbits (SSO)
+        n_sunsync = 1.991063853e-7 * u.one / u.s
+        R_earth = Earth.R
+        k_earth = Earth.k
+        J2_earth = Earth.J2
```

Comment:
As I discussed with @Meuge privately a while ago:

> "Sun-synchronous orbits are possible around other oblate planets, such as Mars. But Venus, for example, is too spherical to have a satellite in sun-synchronous orbit. If 𝐽2 is too small, you can't get sufficient torque to rotate the orbit fast enough. That is the case for Venus, whose 𝐽2 is about 0.4% that of Earth's.  this lack of oblateness is due to Venus' extremely low rotation rate"

[Book](https://www.springer.com/gp/book/9783319034157)

Therefore, where possible I think we should consider raising an error?